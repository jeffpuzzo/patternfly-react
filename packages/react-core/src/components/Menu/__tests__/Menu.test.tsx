import React from 'react';
import { render, screen } from '@testing-library/react';
import { Menu } from '../Menu';

describe('Menu', () => {
  test('should render Menu successfully', () => {
    render(
      <Menu activeItemId={0} onSelect={jest.fn()}>
        content
      </Menu>
    );
    expect(screen.getByText('content').outerHTML).toMatchSnapshot();
  });

  describe('with isPlain', () => {
    test('should render Menu with plain styles applied', () => {
      render(
        <Menu activeItemId={0} onSelect={jest.fn()} isPlain>
          content
        </Menu>
      );
      expect(screen.getByText('content').className).toContain('pf-m-plain');
    });
  });

  describe('with isScrollable', () => {
    test('should render Menu with scrollable styles applied', () => {
      render(
        <Menu activeItemId={0} onSelect={jest.fn()} isScrollable>
          content
        </Menu>
      );
      expect(screen.getByText('content').className).toContain('pf-m-scrollable');
    });
  });

  describe('with isNavFlyout', () => {
    test('should render Menu with nav flyout styles applied', () => {
      render(
        <Menu activeItemId={0} onSelect={jest.fn()} isNavFlyout>
          content
        </Menu>
      );
      expect(screen.getByText('content').className).toContain('pf-m-nav');
    });
  });
});
