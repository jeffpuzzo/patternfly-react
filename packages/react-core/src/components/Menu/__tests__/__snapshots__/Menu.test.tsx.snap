// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render Menu successfully 1`] = `"<div class=\\"pf-c-menu\\" data-ouia-component-type=\\"PF4/Menu\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Menu-1\\">content</div>"`;
